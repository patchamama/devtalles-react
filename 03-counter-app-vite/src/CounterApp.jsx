//https://cursos.devtalles.com/courses/take/react-cero-experto/lessons/36057304-tarea-componente-counterapp
//https://cursos.devtalles.com/courses/take/react-cero-experto/lessons/36057309-evento-click-eventos-en-general
//https://es.legacy.reactjs.org/docs/events.html
//https://cursos.devtalles.com/courses/take/react-cero-experto/lessons/36057314-usestate-hook

import { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value)

  const handleEvent = (event) => {
    //   setCounter(counter + 1)
    setCounter((c) => c + 1)
  }
  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>

      <button onClick={handleEvent}>+1</button>
      <button onClick={(event) => handleEvent(event)}>+1</button>
      <button onClick={() => setCounter(counter + 1)}>+1</button>
    </>
  )
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp
