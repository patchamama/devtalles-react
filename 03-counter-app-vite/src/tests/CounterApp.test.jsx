import { render, screen } from '@testing-library/react'
import CounterApp from '../CounterApp'

describe('Pruebas en el archivo <CounterApp />', () => {
  const initialValue = 10
  test('debe de hacer match con el snapshot', () => {
    const { container } = render(<CounterApp value={initialValue} />)
    expect(container).toMatchSnapshot()
  })

  test('debe de mostrar el valor inicial de 100', () => {
    render(<CounterApp value={100} />)
    expect(screen.getByText('100')).toBeTruthy()

    expect(screen.getByText('100').tagName).toBe('H2')
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain('100')
  })
})
