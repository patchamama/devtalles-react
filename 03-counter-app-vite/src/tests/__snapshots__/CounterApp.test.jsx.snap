// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el archivo <CounterApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
